src/components/Order/Order.js

```diff
@@ -1,92 +1,83 @@
-import React, { Component } from 'react';
-import OrderContext from '../../context/OrderContext'
-import OrderList from '../OrderList/OrderList'
-
-
+import React, { Component } from "react";
+import OrderContext from "../../context/OrderContext";
+import OrderList from "../OrderList/OrderList";
+import { BTC_USD, XRP_USD } from "../../utils/constants";
+import {
+  connectSocket,
+  subscribe,
+  unsubscribe
+} from "../../services/api-service";
+const DATA_EVENT = "data";
 export default class Order extends Component {
-    static contextType = OrderContext;
+  static contextType = OrderContext;
 
-    constructor(){
-        super()
-        this.state = {
-            book: '',
-            error: null,
-            currencies: [{value: "btcusd", option: "BTC/USD"}, {value: "xrpusd", option: "XRP/USD"}]
-        }
+  constructor() {
+    super();
+    this.state = {
+      book: "",
+      error: null,
+      currencies: [
+        { value: BTC_USD.value, option: BTC_USD.option },
+        { value: XRP_USD.value, option: XRP_USD.option }
+      ],
+      ordersData: null
+    };
+  }
 
-    }
-    
-    // ws = new WebSocket("wss://ws.bitstamp.net")
+  componentDidMount() {
+    this.setState({
+      book: "",
+      error: null
+    });
+    connectSocket(this.onMessageHandler);
```

Comment:
we pass the messageHandler here so we get a hold of the values when the data comes from the server.

but probably the best way for the api-server to put the data in the state using Context API and this component will then read the data on the state when it changes